pandas/core/generic.py

```diff
@@ -9974,11 +9974,15 @@ def _agg_by_level(self, name, axis=0, level=0, skipna=True, **kwargs):
             raise ValueError("Must specify 'axis' when aggregating by level.")
         grouped = self.groupby(level=level, axis=axis, sort=False)
         if hasattr(grouped, name) and skipna:
-            return getattr(grouped, name)(**kwargs)
-        axis = self._get_axis_number(axis)
-        method = getattr(type(self), name)
-        applyf = lambda x: method(x, axis=axis, skipna=skipna, **kwargs)
-        return grouped.aggregate(applyf)
+            result = getattr(grouped, name)(**kwargs)
+        else:
+            axis = self._get_axis_number(axis)
+            method = getattr(type(self), name)
+            applyf = lambda x: method(x, axis=axis, skipna=skipna, **kwargs)
+            result = grouped.aggregate(applyf)
+        if isinstance(self, ABCSeries) and self.dtype.name == "boolean":
```

Comment:
can you instead trace exactly where this is not getting converted propertly; likely its in EA method itself. We can to handle much lower down rather than a specific case here

Reply:
In pandas/core/groupby/groupby.py file, there is a _get_cythonized_result() function which has takes a pre_processing lambda where the return type of result is decided.
We can have an if/else statement there which checks the dtype of input and decides the dtype of the result.